service/test/ServiceIOGroupTest.cpp

```diff
@@ -181,6 +181,37 @@ TEST_F(ServiceIOGroupTest, read_signal_behavior)
                                "BAD SIGNAL not valid for ServiceIOGroup");
 }
 
+TEST_F(ServiceIOGroupTest, read_signal_exception)
+{
+    // !is_valid_signal(signal_name)
+    GEOPM_EXPECT_THROW_MESSAGE(
+        m_serviceio_group->read_signal("NUM_VACCUM_TUBES", 4, 0),
```

Comment:
```suggestion
        m_serviceio_group->read_signal("NUM_VACUUM_TUBES", 4, 0),
```

I know it's a nit, but it's spelled "vacuum".  This does not affect the functionality.